TestGetLoadBalancer race detected during execution of test

Since #62565 merged on Friday night, my PR (#62572) has failed several retests of pull-kubernetes-bazel-test. My PR is unrelated to the load balancer so I suspect that #62565 introduced a flake. I see several runs of that test against master that aren't failing, so clearly the test works sometimes (and the original PR passed it several times to before it merged). Perhaps the test harness sometimes runs problem tests in parallel and other times it doesn't.

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:

My PR had one good run of pull-kubernetes-bazel-test before #62565 merged, but has failed several times since with no code changes to my PR.

```
E0415 18:50:24.388907      15 gce_loadbalancer_external.go:461] verifyUserRequestedIP: requested IP "1.1.1.1" for LB test-lb is neither static nor assigned to the LB
E0415 18:50:24.391401      15 gce_loadbalancer_external.go:445] verifyUserRequestedIP: requested static IP "1.1.1.1" (name: foo) for LB test-lb has network tier Premium, need Standard.
E0415 18:50:24.410287      15 gce_loadbalancer_external.go:75] ensureExternalLoadBalancer(a(/)): Failed to get the desired network tier: unsupported network tier: "SupremeLuxury".
E0415 18:50:24.480914      15 gce_healthchecks.go:195] vstring () is not a valid version string: could not parse "" as version
E0415 18:50:24.481083      15 gce_instances.go:460] Failed to retrieve instances: []
==================
WARNING: DATA RACE
Read at 0x000002fb6c08 by goroutine 61:
  runtime.convT2E64()
      GOROOT/src/runtime/iface.go:335 +0x0
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.convertAndInsertAlphaAddress()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:200 +0xc64
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.InsertAddressHook()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:216 +0x13f
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud.(*MockAddresses).Insert()
      pkg/cloudprovider/providers/gce/cloud/gen.go:1309 +0xbd2
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ReserveRegionAddress()
      pkg/cloudprovider/providers/gce/gce_addresses.go:67 +0x1fd
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.ensureStaticIP()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go:986 +0x803
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ensureExternalLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go:148 +0x37c8
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).EnsureLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer.go:149 +0x9dc
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.TestEnsureLoadBalancerCreatesExternalLb()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_test.go:66 +0x554
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Previous write at 0x000002fb6c08 by goroutine 86:
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.convertAndInsertAlphaAddress()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:201 +0xd6a
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.InsertAddressHook()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:216 +0x13f
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud.(*MockAddresses).Insert()
      pkg/cloudprovider/providers/gce/cloud/gen.go:1309 +0xbd2
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ReserveRegionAddress()
      pkg/cloudprovider/providers/gce/gce_addresses.go:67 +0x1fd
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.ensureStaticIP()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go:986 +0x803
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ensureExternalLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go:148 +0x37c8
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.createExternalLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go:277 +0x15e
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.TestEnsureLoadBalancerDeletedDeletesExternalLb()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_test.go:147 +0x502
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Goroutine 61 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d
  testing.runTests()
      GOROOT/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:978 +0x2cd
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/cloudprovider/providers/gce/linux_amd64_race_stripped/go_default_test~/testmain.go:267 +0x3ad

Goroutine 86 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d
  testing.runTests()
      GOROOT/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:978 +0x2cd
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/cloudprovider/providers/gce/linux_amd64_race_stripped/go_default_test~/testmain.go:267 +0x3ad
==================
==================
WARNING: DATA RACE
Read at 0x000002fb6c08 by goroutine 84:
  runtime.convT2E64()
      GOROOT/src/runtime/iface.go:335 +0x0
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.convertAndInsertAlphaAddress()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:200 +0xc64
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.InsertAddressHook()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:216 +0x13f
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud.(*MockAddresses).Insert()
      pkg/cloudprovider/providers/gce/cloud/gen.go:1309 +0xbd2
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ReserveRegionAddress()
      pkg/cloudprovider/providers/gce/gce_addresses.go:67 +0x1fd
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*addressManager).ensureAddressReservation()
      pkg/cloudprovider/providers/gce/gce_address_manager.go:130 +0x36d
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*addressManager).HoldAddress()
      pkg/cloudprovider/providers/gce/gce_address_manager.go:93 +0x2ec
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ensureInternalLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_internal.go:98 +0x1862
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.createInternalLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_internal_test.go:41 +0x162
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.TestEnsureLoadBalancerDeletesExistingInternalLb()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_test.go:102 +0x559
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Previous write at 0x000002fb6c08 by goroutine 60:
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.convertAndInsertAlphaAddress()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:201 +0xd6a
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud/mock.InsertAddressHook()
      pkg/cloudprovider/providers/gce/cloud/mock/mock.go:216 +0x13f
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce/cloud.(*MockAddresses).Insert()
      pkg/cloudprovider/providers/gce/cloud/gen.go:1309 +0xbd2
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ReserveRegionAddress()
      pkg/cloudprovider/providers/gce/gce_addresses.go:67 +0x1fd
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.ensureStaticIP()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go:986 +0x803
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).ensureExternalLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go:148 +0x37c8
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.(*GCECloud).EnsureLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer.go:149 +0x9dc
  k8s.io/kubernetes/pkg/cloudprovider/providers/gce.TestGetLoadBalancer()
      pkg/cloudprovider/providers/gce/gce_loadbalancer_test.go:45 +0x697
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d

Goroutine 84 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d
  testing.runTests()
      GOROOT/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:978 +0x2cd
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/cloudprovider/providers/gce/linux_amd64_race_stripped/go_default_test~/testmain.go:267 +0x3ad

Goroutine 60 (running) created at:
  testing.(*T).Run()
      GOROOT/src/testing/testing.go:824 +0x564
  testing.runTests.func1()
      GOROOT/src/testing/testing.go:1063 +0xa4
  testing.tRunner()
      GOROOT/src/testing/testing.go:777 +0x16d
  testing.runTests()
      GOROOT/src/testing/testing.go:1061 +0x4e1
  testing.(*M).Run()
      GOROOT/src/testing/testing.go:978 +0x2cd
  main.main()
      bazel-out/k8-fastbuild/bin/pkg/cloudprovider/providers/gce/linux_amd64_race_stripped/go_default_test~/testmain.go:267 +0x3ad
==================
--- FAIL: TestEnsureLoadBalancerDeletedDeletesInternalLb (0.00s)
	testing.go:730: race detected during execution of test
--- FAIL: TestGetLoadBalancer (0.00s)
	testing.go:730: race detected during execution of test
--- FAIL: TestEnsureLoadBalancerCreatesInternalLb (0.01s)
	testing.go:730: race detected during execution of test
--- FAIL: TestEnsureLoadBalancerCreatesExternalLb (0.01s)
	testing.go:730: race detected during execution of test
--- FAIL: TestEnsureLoadBalancerDeletedDeletesExternalLb (0.01s)
	testing.go:730: race detected during execution of test
--- FAIL: TestEnsureLoadBalancerDeletesExistingExternalLb (0.01s)
	testing.go:730: race detected during execution of test
--- FAIL: TestEnsureLoadBalancerDeletesExistingInternalLb (0.01s)
	testing.go:730: race detected during execution of test
FAIL
```

**What you expected to happen**:

The tests should consistently pass unless related problem code is introduced.

--- a/pkg/cloudprovider/providers/gce/gce_loadbalancer_test.go
+++ b/pkg/cloudprovider/providers/gce/gce_loadbalancer_test.go
@@ -25,8 +25,6 @@ import (
 )
 
 func TestGetLoadBalancer(t *testing.T) {
-	t.Parallel()
-
 	vals := DefaultTestClusterValues()
 	gce, err := fakeGCECloud(vals)
 	require.NoError(t, err)
@@ -52,8 +50,6 @@ func TestGetLoadBalancer(t *testing.T) {
 }
 
 func TestEnsureLoadBalancerCreatesExternalLb(t *testing.T) {
-	t.Parallel()
-
 	vals := DefaultTestClusterValues()
 	gce, err := fakeGCECloud(vals)
 	require.NoError(t, err)
@@ -70,8 +66,6 @@ func TestEnsureLoadBalancerCreatesExternalLb(t *testing.T) {
 }
 
 func TestEnsureLoadBalancerCreatesInternalLb(t *testing.T) {
-	t.Parallel()
-
 	vals := DefaultTestClusterValues()
 	gce, err := fakeGCECloud(vals)
 	require.NoError(t, err)
@@ -88,8 +82,6 @@ func TestEnsureLoadBalancerCreatesInternalLb(t *testing.T) {
 }
 
 func TestEnsureLoadBalancerDeletesExistingInternalLb(t *testing.T) {
-	t.Parallel()
-
 	vals := DefaultTestClusterValues()
 	gce, err := fakeGCECloud(vals)
 	require.NoError(t, err)
@@ -110,8 +102,6 @@ func TestEnsureLoadBalancerDeletesExistingInternalLb(t *testing.T) {
 }
 
 func TestEnsureLoadBalancerDeletesExistingExternalLb(t *testing.T) {
-	t.Parallel()
-
 	vals := DefaultTestClusterValues()
 	gce, err := fakeGCECloud(vals)
 	require.NoError(t, err)
@@ -133,8 +123,6 @@ func TestEnsureLoadBalancerDeletesExistingExternalLb(t *testing.T) {
 }
 
 func TestEnsureLoadBalancerDeletedDeletesExternalLb(t *testing.T) {
-	t.Parallel()
-
 	vals := DefaultTestClusterValues()
 	gce, err := fakeGCECloud(vals)
 	require.NoError(t, err)
@@ -152,8 +140,6 @@ func TestEnsureLoadBalancerDeletedDeletesExternalLb(t *testing.T) {
 }
 
 func TestEnsureLoadBalancerDeletedDeletesInternalLb(t *testing.T) {
-	t.Parallel()
-
 	vals := DefaultTestClusterValues()
 	gce, err := fakeGCECloud(vals)
 	require.NoError(t, err)

